django/contrib/auth/forms.py

```diff
@@ -82,7 +82,9 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super(UserCreationForm, self).__init__(*args, **kwargs)
-        self.fields['username'].widget.attrs.update({'autofocus': ''})
+        username_field = self.fields.get('username')
```

Comment:
I was thinking something similar to https://github.com/django/django/blob/master/django/contrib/auth/forms.py#L158, but your suggestion is better. Done, thanks!

I also removed the if branch. Perhaps we could catch the KeyError and raise an exception with a more informed message (e.g. USERNAME_FIELD cannot be found in FooCreationForm.Meta.fields)?